dotcom-rendering/scripts/env/check-deps.js

```diff
@@ -1,9 +1,33 @@
+const fs = require('fs');
+const lockfile = require('@yarnpkg/lockfile');
 const pkg = require('../../package.json');
+const { warn, log } = require('./log');
 
 if (pkg.devDependencies) {
-	const { warn, log } = require('./log');
-
-	warn('Don’t use devDependencies');
+	warn('Don’t use dependencies');
 	log('See https://github.com/guardian/dotcom-rendering/pull/4001');
 	process.exit(1);
 }
+
+/** @type {{ object: import('@types/yarnpkg__lockfile').LockFileObject }}*/
+const { object: json } = lockfile.parse(
+	fs.readFileSync('../yarn.lock', 'utf8'),
+);
+
+const knownNonSemver = /** @type {const} */ ([
+	'https://github.com/guardian/babel-plugin-px-to-rem#v0.1.0',
+]);
+
+const mismatches = Object.entries(pkg.dependencies)
+	.filter(([name, version]) => {
+		const pinned = json[name + '@' + version]?.version;
+		return version !== pinned;
+	})
+	.filter(([, version]) => !knownNonSemver.includes(version));
+
+if (mismatches.length) warn('All dependencies should be pinned');
+for (const [name, version] of mismatches) {
+	warn(`You must fix: ${name}@${String(version)}`);
+}
+
+if (mismatches.length !== 0) process.exit(mismatches.length);
```

Comment:
```suggestion
if (mismatches.length !== 0) process.exit(1);
```
`exit` takes an exit code as a parameter, and a failed script should exit with `1` regardless of how many mismatches there are

Reply:
My understanding was that [any nonzero code would do](https://en.wikipedia.org/wiki/Exit_status#Shell_and_scripts).

Reply:
I think some exit codes have special meanings (e.g. `2` means "Misuse of shell builtins"). `3-126` are unreserved and then they start having a meaning again,  see https://tldp.org/LDP/abs/html/exitcodes.html
I normally always resort to a helpful error message + exit code 1 when I want to indicate some sort of failure to avoid any confusion with exit codes, but I defer to someone with more knowledge than me in this field!